specifications/api/swagger.yaml

```diff
@@ -1474,6 +1525,37 @@ components:
           passphrase: *lenientPassphrase
           metadata: *transactionMetadata
 
+    ApiScript: &ApiScript
+      type: object
+      required:
+        - script
+      properties:
+        script: *script
+
+    ApiPubKey: &ApiPubKey
+      type: object
+      required:
+        - pub_key
+      properties:
+        pub_key: *credentialPubKey
+
+    AnyAddress: &AnyAddress
+      type: object
+      required:
+        - address
+      properties:
+        address: *anyAddress
+
+    ApiCredential: &ApiCredential
```

Comment:
@rvl we typically keep the same naming as in the `Cardano.Wallet.Api.Types` module for the swagger file references because it creates a straightforward mapping for the automated schema validations we perform in tests. This way, one can write `ToSchema` instances as:

```hs
instance ToSchema (ApiSelectCoinsPayments n) where
    declareNamedSchema _ = declareSchemaForDefinition "ApiSelectCoinsPayments"

instance ToSchema ApiSelectCoinsAction where
    declareNamedSchema _ = declareSchemaForDefinition "ApiSelectCoinsAction"

instance ToSchema (ApiCoinSelection n) where
    declareNamedSchema _ = declareSchemaForDefinition "ApiCoinSelection"

instance ToSchema ApiWallet where
    declareNamedSchema _ = declareSchemaForDefinition "ApiWallet"

instance ToSchema ApiByronWallet where
    declareNamedSchema _ = declareSchemaForDefinition "ApiByronWallet"
```

Reply:
We could even push it further and have default instances relying on generics and Typeable.

Reply:
Aha ok thanks.